tests/artifacts_test.go

```diff
@@ -0,0 +1,186 @@
+			ArtifactType string `json:"artifact_type"`
+		}
+		type ResponseArtifactData struct {
+			Artifacts []ResponseArtifact `json:"artifacts"`
+		}
+
+		name := "Test Iris Pipeline"
+		uploadUrl := fmt.Sprintf("%s/apis/v2beta1/pipelines/upload?name=%s", APIServerURL, url.QueryEscape(name))
+		log.Printf("Uploading pipeline: %s to URL: %s", name, uploadUrl)
+
+		vals := map[string]string{
+			"uploadfile": "@resources/iris_pipeline_without_cache_compiled.yaml",
+		}
+		bodyUpload, contentTypeUpload := TestUtil.FormFromFile(t, vals)
+		response, err := suite.Clientmgr.httpClient.Post(uploadUrl, contentTypeUpload, bodyUpload)
+		require.NoError(t, err, "Failed to upload pipeline")
+		responseData, err := io.ReadAll(response.Body)
+		require.NoError(t, err, "Failed to read response data")
+		require.Equal(t, http.StatusOK, response.StatusCode, "Unexpected HTTP status code")
+		log.Println("Pipeline uploaded successfully.")
+
+		// Retrieve Pipeline ID
+		log.Println("Retrieving Pipeline ID...")
+		pipelineID, err := TestUtil.RetrievePipelineId(t, suite.Clientmgr.httpClient, APIServerURL, name)
+		require.NoError(t, err, "Failed to retrieve Pipeline ID")
+		log.Printf("Pipeline ID: %s", pipelineID)
+
+		// Create a new run
+		log.Println("Creating a new pipeline run...")
+		runUrl := fmt.Sprintf("%s/apis/v2beta1/runs", APIServerURL)
+		bodyRun := TestUtil.FormatRequestBody(t, pipelineID, name)
+		contentTypeRun := "application/json"
+		response, err = suite.Clientmgr.httpClient.Post(runUrl, contentTypeRun, bytes.NewReader(bodyRun))
+		require.NoError(t, err, "Failed to create pipeline run")
+		responseData, err = io.ReadAll(response.Body)
+		require.NoError(t, err, "Failed to read run response data")
+		require.Equal(t, http.StatusOK, response.StatusCode, "Unexpected HTTP status code")
+		runID := TestUtil.RetrieveRunID(t, responseData)
+		log.Println("Pipeline run created successfully.")
+
+		err = TestUtil.WaitForPipelineRunCompletion(t, suite.Clientmgr.httpClient, APIServerURL)
+		require.NoError(t, err, "Pipeline run did not complete successfully")
+
+		// Fetch artifacts from API
+		artifactsUrl := fmt.Sprintf("%s/apis/v2beta1/artifacts?run_id=%s&namespace=%s", APIServerURL, runID, suite.DSPANamespace)
+		log.Printf("Fetching artifacts from URL: %s", artifactsUrl)
+		response, err = suite.Clientmgr.httpClient.Get(artifactsUrl)
+		require.NoError(t, err, "Failed to fetch artifacts")
+		responseData, err = io.ReadAll(response.Body)
+		require.NoError(t, err, "Failed to read artifacts response data")
+		require.Equal(t, http.StatusOK, response.StatusCode, "Unexpected HTTP status code")
+
+		// Parse the artifact list
+		var responseArtifactsData struct {
+			Artifacts []ResponseArtifact `json:"artifacts"`
+		}
+		err = json.Unmarshal(responseData, &responseArtifactsData)
+		require.NoError(t, err, "Failed to parse artifacts response JSON")
+
+		for _, artifact := range responseArtifactsData.Artifacts {
+			if artifact.ArtifactType != "system.Model" {
```

Comment:
I think we can also include datasets:
```suggestion
			if artifact.ArtifactType != "system.Model" && artifact.ArtifactType != "sytem.Dataset" {
```